wandb/sdk/artifacts/artifact_saver.py

```diff
@@ -72,9 +72,9 @@ def save(
         incremental: bool = False,
         history_step: Optional[int] = None,
         base_id: Optional[str] = None,
-    ) -> Optional[Dict]:
+    ) -> Tuple[Optional[Dict], Optional[concurrent.futures.Future[None]]]:
```

Comment:
This might be causing your test failures. I saw this in one of the logs: 
![image](https://github.com/wandb/wandb/assets/14353406/23773a80-949f-4374-9fa4-95461ef20cc6)

```suggestion
    ) -> Tuple[Optional[Dict], Optional[concurrent.futures.Future]]:
```

Reply:
fixed